plugins/ocm-backend/src/helpers/config.ts

```diff
@@ -1,29 +1,65 @@
 import { Config } from '@backstage/config';
+import { Logger } from 'winston';
 
 const CLUSTERS_PATH = 'kubernetes.clusterLocatorMethods';
-const HUB_CLUSTER_CONFIG_PATH = 'ocm.hub';
+const OCM_PREFIX = 'ocm';
+const CLUSTER_CONFIG_PATH = `${OCM_PREFIX}.cluster`;
+const HUB_CONFIG_PATH = `${OCM_PREFIX}.hub`;
 
-export const getHubClusterName = (config: Config) =>
-  config.getString(HUB_CLUSTER_CONFIG_PATH);
-
-export const getHubClusterFromConfig = (config: Config) => {
+export const configHandler = (
+  config: Config | undefined,
+  prefix: string,
+  key: string,
+) => {
   try {
-    getHubClusterName(config);
+    return config!.getString(key);
   } catch (err) {
     throw new Error(
-      `Hub cluster must be specified in config at '${HUB_CLUSTER_CONFIG_PATH}'`,
+      `Hub cluster ${key} must be specified in config at '${prefix}.${key}'`,
+    );
+  }
+};
+
+export const getHubClusterName = (config: Config): string => {
+  const hubConfig = config.getOptionalConfig(HUB_CONFIG_PATH);
+  const clusterConfig = config.getOptionalConfig(CLUSTER_CONFIG_PATH);
+  if (!hubConfig && !clusterConfig) {
+    throw new Error(
+      `Neither hub or cluster configuration were specified at '${OCM_PREFIX}' config`,
     );
   }
 
+  return hubConfig
+    ? configHandler(hubConfig, HUB_CONFIG_PATH, 'name')
+    : configHandler(clusterConfig, CLUSTER_CONFIG_PATH, 'name');
+};
+
+export const getHubClusterFromKubernetesConfig = (config: Config): Config => {
+  // Check if required value is valid
+  getHubClusterName(config);
```

Comment:
Why would we do this here? We should call this function only if we already know the user chose `ocm.cluster` way. What's the reason for this call? Like.. is an attempt to raise error on invalid config?

Reply:
This was stupid on my part, fixed in the latest commit by getting the name straight from the `ocm.cluster` config.